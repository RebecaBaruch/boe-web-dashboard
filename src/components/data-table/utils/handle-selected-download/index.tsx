import { pdf } from '@react-pdf/renderer';
import {
  PDFDocumentGenerator,
  PDFDocumentProps,
} from '../../../../../src/utils/pdf-document-generator';

const downloadPDF = async (
  record: PDFDocumentProps,
  fileName: string = 'diagnosis_report.pdf',
) => {
  const pdfDocument = PDFDocumentGenerator(record) as React.ReactElement;

  const blob = await pdf(pdfDocument).toBlob();

  const link = document.createElement('a');
  link.href = URL.createObjectURL(blob);
  link.download = fileName;
  link.click();
};

export default downloadPDF;
