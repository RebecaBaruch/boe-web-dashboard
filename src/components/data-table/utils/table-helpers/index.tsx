import React from 'react';
import { Checkbox } from '@telefonica/mistica';
import { ActionButton, ActionModal, TableCell } from '../../styled';
import { MoreVertical } from 'react-feather';
import ActionButtons from 'components/data-table/action-buttons/action-buttons.component';

import {
  TableAccountsData,
  TableDiagnosisData,
  TableAnimalsData,
  Column,
} from '../../types';

interface SelectableCellProps {
  isSelected: boolean;
  onSelect: () => void;
}

interface ActionCellProps {
  onActionClick: () => void;
}

interface RenderActionCellProps {
  action?: 'download' | 'delete';
  selectMode?: boolean;
  isSelected: boolean;
  toggleSelectRow?: (id: string) => void;
  item: TableAccountsData | TableDiagnosisData | TableAnimalsData;
  rowIndex: number;
  onAction: (rowIndex: number) => void;
}

interface RenderActionModalProps {
  activeRowIndex: number | null;
  rowIndex: number;
  action?: 'download' | 'delete';
  modalRef: React.RefObject<HTMLDivElement>;
  data: TableAccountsData[] | TableDiagnosisData[] | TableAnimalsData[];
}

export const renderCell = (
  col: Column,
  item: TableAccountsData | TableDiagnosisData | TableAnimalsData,
): React.ReactNode => {
  if (col.render) {
    return col.render(item);
  }

  return item[col.accessor as keyof typeof item];
};

export const SelectableCell: React.FC<SelectableCellProps> = ({
  isSelected,
  onSelect,
}) => (
  <TableCell>
    <Checkbox name="select" checked={isSelected} onChange={onSelect} />
  </TableCell>
);

export const ActionCell: React.FC<ActionCellProps> = ({ onActionClick }) => (
  <TableCell>
    <ActionButton onClick={onActionClick}>
      <MoreVertical width={17} height={17} />
    </ActionButton>
  </TableCell>
);

export const getIsSelected = (
  selectedRowsSet: Set<string> | string[],
  itemId: string,
): boolean => {
  if (selectedRowsSet instanceof Set) {
    return selectedRowsSet.has(itemId); // Usando has para Set
  }
  return (selectedRowsSet as string[]).includes(itemId); // Usando includes para Array
};

export const renderActionCell = ({
  action,
  selectMode,
  isSelected,
  toggleSelectRow,
  item,
  rowIndex,
  onAction,
}: RenderActionCellProps) => {
  if (!action) return null;

  return selectMode ? (
    <SelectableCell
      isSelected={isSelected}
      onSelect={() => toggleSelectRow?.(item.id)}
    />
  ) : (
    <ActionCell onActionClick={() => onAction(rowIndex)} />
  );
};

export const renderActionModal = ({
  activeRowIndex,
  rowIndex,
  action,
  modalRef,
  data,
}: RenderActionModalProps) => {
  if (activeRowIndex !== rowIndex) return null;

  return (
    <ActionModal ref={modalRef}>
      <ActionButtons
        action={action}
        onDelete={() => {}}
        activeRowIndex={activeRowIndex}
        data={data}
      />
    </ActionModal>
  );
};
