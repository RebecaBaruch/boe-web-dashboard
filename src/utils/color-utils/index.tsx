// function to get the color of the risk based on the value
export const getRiskColor = (risk: number): string => {
  if (risk >= 80) {
    return '#FF5454';
  } else if (risk >= 40) {
    return '#FFEB36';
  } else {
    return '#3BD640';
  }
};

export const getRiskTextColor = (risk: number): string => {
  if (risk >= 40 && risk < 80) {
    return '#474000';
  } else {
    return '#FFFFFF';
  }
};

// function to get the color of the result based on the value
export const getResultColor = (result: string): string => {
  switch (result) {
    case 'Berne':
      return '#4A65CF';
    case 'Dermatite':
      return '#00DAB6';
    case 'Dermatofitose':
      return '#FF12CB';
    case 'Dermatofilose':
      return '#8515E8';
    default:
      return '#ffffff';
  }
};

export const getResultTextColor = (result: string): string => {
  switch (result) {
    case 'Dermatite':
      return '#001F3C';
    case null:
      return '#717171';
    default:
      return '#ffffff';
  }
};

export const getStatusColor = (status: string): string => {
  switch (status) {
    case 'Em andamento':
      return '#FFAD0F';
    case 'Concluído':
      return '#45A6FF';
    default:
      return '#F2F2F5';
  }
};

export const getStatusTextColor = (status: string): string => {
  switch (status) {
    case 'Em andamento':
      return '#4C3200';
    case 'Concluído':
      return '#FFFFFF';
    default:
      return '#717171';
  }
};
