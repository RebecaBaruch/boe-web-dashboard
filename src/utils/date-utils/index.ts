// Returns the current day of the week in Portuguese
export function getCurrentDayOfWeek(): string {
  const daysOfWeek = [
    'Domingo',
    'Segunda-feira',
    'Terça-feira',
    'Quarta-feira',
    'Quinta-feira',
    'Sexta-feira',
    'Sábado',
  ];
  const today = new Date();
  return daysOfWeek[today.getDay()];
}

// Returns the current date in the format "31 Oct, 2024"
export function getCurrentDate(): string {
  const today = new Date();
  const day = today.getDate();
  const months = [
    'Jan',
    'Feb',
    'Mar',
    'Apr',
    'May',
    'Jun',
    'Jul',
    'Aug',
    'Sep',
    'Oct',
    'Nov',
    'Dec',
  ];
  const month = months[today.getMonth()];
  const year = today.getFullYear();

  return `${day} ${month}, ${year}`;
}

// Returns the current time in the format "18h30"
export function getCurrentTime(): string {
  const now = new Date();
  const hours = now.getHours().toString().padStart(2, '0');
  const minutes = now.getMinutes().toString().padStart(2, '0');

  return `${hours}h${minutes}`;
}