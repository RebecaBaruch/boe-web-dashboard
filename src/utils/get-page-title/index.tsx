import { RoutesUrls } from "../enums/routes-url";

export const getPageTitle = (pathname: string): string => {
  const titles: Record<string, string> = {
    [RoutesUrls.BASE_URL]: 'Dashboard',
    [RoutesUrls.LINKED_ACCOUNTS]: 'Contas associadas',
    [RoutesUrls.DIAGNOSIS_HISTORIC]: 'Histórico de Análises',
    [RoutesUrls.USER_PROFILE]: 'Meu perfil',
    [RoutesUrls.LOGIN]: 'Login',
    [RoutesUrls.REGISTER]: 'Registro',
    [RoutesUrls.REGISTER_FARM]: 'Registro - Fazenda',
  };
  return titles[pathname] || '';
};