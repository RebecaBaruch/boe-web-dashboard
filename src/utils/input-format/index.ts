export const formatPhone = (value: string) => {
  const onlyNumbers = value.replace(/\D/g, '');

  const maxLength = 11;

  if (onlyNumbers.length == maxLength) {
    return `(${onlyNumbers.slice(0, 2)}) ${onlyNumbers.slice(2, 7)}-${onlyNumbers.slice(7, 11)}`;
  }
};

export const formatCep = (value: string) => {
  const onlyNumbers = value.replace(/\D/g, '');

  const maxLength = 8;

  if (onlyNumbers.length == maxLength) {
    return `${onlyNumbers.slice(0, 5)}-${onlyNumbers.slice(5, 8)}`;
  }
};

export const formatCnpj = (value: string) => {
  const onlyNumbers = value.replace(/\D/g, '');

  const maxLength = 14;

  if (onlyNumbers.length == maxLength) {
    return `${onlyNumbers.slice(0, 2)}.${onlyNumbers.slice(2, 5)}.${onlyNumbers.slice(5, 8)}/${onlyNumbers.slice(8, 12)}-${onlyNumbers.slice(12, 14)}`;
  }
};

export const togglePasswordVisibility = (
  isVisible: boolean,
  value: string,
): { inputType: string; displayValue: string } => {
  return {
    inputType: isVisible ? 'text' : 'password',
    displayValue: isVisible ? value : '*'.repeat(value.length), // Oculta a senha com asteriscos
  };
};

export const isValidEmail = (email: string | undefined) => {
  if (!email) return false;

  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return emailRegex.test(email);
};
