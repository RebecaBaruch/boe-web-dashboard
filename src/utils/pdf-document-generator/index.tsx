import React from 'react';
import {
  Document,
  Page,
  Text,
  View,
  Image,
  StyleSheet,
} from '@react-pdf/renderer';
import { TableDiagnosisData } from 'components/data-table/types';
import {
  getResultColor,
  getResultTextColor,
  getStatusColor,
  getStatusTextColor,
} from 'utils/color-utils';

const styles = StyleSheet.create({
  page: {
    fontFamily: 'Helvetica',
    padding: 20,
  },
  header: {
    fontSize: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 20,
  },
  sectionTitle: {
    fontSize: 14,
    fontWeight: 'bold',
    marginBottom: 10,
  },
  animalPhoto: {
    width: 40,
    height: 40,
    borderRadius: 7,
  },
  animalInfoContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  analysisContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 20,
    borderBottom: '1 solid #00000034',
  },
  resultContainer: {
    flexDirection: 'row',
    gap: 20,
    marginBottom: 20,
  },
  resultBox: {
    flexDirection: 'column',
    gap: 5,
  },
  descriptionContainer: {
    flexDirection: 'row',
    gap: 20,
    borderRadius: 10,
    border: '1 solid #00000034',
  },
  descriptionBox: {
    flexDirection: 'column',
    gap: 5,
  },
  analysisImg: {
    display: 'flex',
    justifyContent: 'center',
    alignContent: 'center',
    marginRight: 32,
    width: 81,
    height: 81,
    borderRadius: 10,
    overflow: 'hidden',
  },
  badge: {
    padding: 4,
    borderRadius: 8,
    fontSize: 12,
    display: 'flex',
    marginBottom: 4,
  },
  authorContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    width: 50,
    padding: '4px 15px',
    borderRadius: 10,
    backgroundColor: '#f2f2f5',
  },
  authorPhoto: {
    width: 30,
    height: 30,
    borderRadius: '50%',
    marginRight: 8,
  },
  authorName: {
    fontSize: 12,
  },
});

interface PDFDocumentProps {
  record: TableDiagnosisData;
}

export const PDFDocumentGenerator: React.FC<PDFDocumentProps> = ({
  record,
}) => {
  const defaultAnimalImg = '../../../public/assets/animal-img.png';
  const defaultAnimalPic = '/public/assets/pic-image.png';

  const resultBadgeStyle = {
    backgroundColor: getResultColor(record.result),
    color: getResultTextColor(record.result),
  };

  const statusBadgeStyle = {
    backgroundColor: getStatusColor(record.status || 'Sem tratamento'),
    color: getStatusTextColor(record.status || 'Sem tratamento'),
  };

  return (
    <Document>
      <Page style={styles.page}>
        <View style={styles.header}>
          <Text>
            <b>Pré-diagnóstico</b>
          </Text>
          <Text>
            <b>Data de registro</b> {record.date}
          </Text>
        </View>

        <View style={styles.analysisContainer}>
          <View style={styles.animalInfoContainer}>
            <Image
              src={record.animalPhoto || defaultAnimalImg}
              style={styles.animalPhoto}
            />
            <View style={{ marginLeft: 10 }}>
              <Text style={{ fontSize: 14, fontWeight: 'bold' }}>
                {record.animalId}
              </Text>
              <Text style={{ fontSize: 12 }}>{record.animalName}</Text>
            </View>
          </View>
          <View style={styles.authorContainer}>
            {record.authorPhoto ? (
              <Image src={record.authorPhoto} style={styles.authorPhoto} />
            ) : (
              <View style={styles.authorPhoto} />
            )}
            <Text style={styles.authorName}>{record.authorName}</Text>
          </View>
        </View>

        <View style={styles.resultContainer}>
          <View style={styles.resultBox}>
            <Text style={{ fontSize: 14, fontWeight: 'bold' }}>Resultado</Text>
            <View style={[styles.badge, resultBadgeStyle]}>
              <Text>{record.result}</Text>
            </View>
          </View>
          <View style={styles.resultBox}>
            <Text style={{ fontSize: 14, fontWeight: 'bold' }}>
              Status de tratamento
            </Text>
            <View style={[styles.badge, statusBadgeStyle]}>
              <Text>{record.status || 'Sem tratamento'}</Text>
            </View>
          </View>
        </View>

        <View style={styles.analysisContainer}>
          <Image
            src={record.analysisImg || defaultAnimalPic}
            style={styles.analysisImg}
          />
          <View style={styles.descriptionBox}>
            <Text style={{ fontSize: 14, fontWeight: 'bold' }}>
              Descrição da análise
            </Text>
            <Text style={{ fontSize: 10, fontWeight: 'light' }}>
              Probabilidade baixa, significa que as lesões ou sintomas
              observados não correspondem aos sinais típicos da doença.
            </Text>
          </View>
        </View>

        <Text style={{ fontSize: 10, color: '#666', marginTop: 30 }}>Böe</Text>
      </Page>
    </Document>
  );
};
