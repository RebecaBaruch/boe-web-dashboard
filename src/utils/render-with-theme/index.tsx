import React from 'react';
import { ThemeContextProvider } from '@telefonica/mistica';
import { render, RenderResult } from '@testing-library/react';
import { theme } from '../../config/theme';

export const renderWithTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeContextProvider theme={theme}>{children}</ThemeContextProvider>,
  );
};
