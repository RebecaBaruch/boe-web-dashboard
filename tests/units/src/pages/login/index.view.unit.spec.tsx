import React from 'react';
import { screen } from '@testing-library/react';
import { renderWithTheme } from '../../../../../src/utils/render-with-theme';
import Login from '../../../../../src/pages/login/index.page';

describe('Login', () => {
  it('should render Login with text Login', () => {
    const { container } = renderWithTheme(<Login />);

    expect(screen.getByText('Log in')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
