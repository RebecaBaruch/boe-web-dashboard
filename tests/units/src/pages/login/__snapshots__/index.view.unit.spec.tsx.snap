// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render Login with text Login 1`] = `
<div
  style="isolation: isolate;"
>
  <div
    class="sc-blHHSb fGQVCr"
  >
    <div
      class="sc-gtLWhw dNridZ"
    >
      <div
        class="_1xgjmkvc"
        style="--_1xgjmkv2: 0px; --_1xgjmkv1: 0px; --_1xgjmkv0: 0px; --_1xgjmkv5: 0px; --_1xgjmkv4: 0px; --_1xgjmkv3: 0px; --_1xgjmkv8: 0px; --_1xgjmkv7: 0px; --_1xgjmkv6: 0px; --_1xgjmkvb: 0px; --_1xgjmkva: 0px; --_1xgjmkv9: 0px; width: 56%; box-sizing: border-box;"
      >
        <div
          class="xjgm6d4"
          style="--xjgm6d0: 32px;"
        >
          <div>
            <img
              alt="boe-logo"
              src="[object Object]"
              width="60"
            />
          </div>
          <div>
            <div
              class="_1xgjmkvc"
              style="--_1xgjmkv2: 0px; --_1xgjmkv1: 0px; --_1xgjmkv0: 0px; --_1xgjmkv5: 0px; --_1xgjmkv4: 0px; --_1xgjmkv3: 0px; --_1xgjmkv8: 0px; --_1xgjmkv7: 0px; --_1xgjmkv6: 0px; --_1xgjmkvb: 0px; --_1xgjmkva: 0px; --_1xgjmkv9: 0px; width: 100%; box-sizing: border-box;"
            >
              <div
                class="xjgm6d4"
                style="--xjgm6d0: 32px;"
              >
                <div>
                  <div
                    class="xjgm6d4"
                    style="--xjgm6d0: 8px;"
                  >
                    <div>
                      <span
                        class="splu5g7 splu5g5"
                        data-component-name="Text5"
                        style="--splu5g0: 1.250rem; --splu5g2: 1.500rem; --splu5g1: 1.750rem; --splu5g3: 2.000rem; hyphens: auto; font-weight: 300; text-transform: inherit; text-decoration: inherit; overflow-wrap: anywhere; white-space: pre-line;"
                      >
                        <div
                          class="title"
                        >
                          Log in
                        </div>
                      </span>
                    </div>
                    <div>
                      <span
                        class="splu5g7 splu5g5"
                        data-component-name="Text1"
                        style="--splu5g0: 0.750rem; --splu5g2: 1.000rem; --splu5g1: 0.875rem; --splu5g3: 1.250rem; hyphens: auto; font-weight: 400; text-transform: inherit; text-decoration: inherit; overflow-wrap: anywhere; white-space: pre-line;"
                      >
                        Bem vindo de volta! Monitore da melhor forma a saúde do seu gado!
                      </span>
                    </div>
                  </div>
                </div>
                <div>
                  <button
                    class="sc-dntaoT hPDAJS"
                    height="3rem"
                  >
                    <div
                      class="button-content"
                    >
                      <img
                        alt="google-logo"
                        src="[object Object]"
                        width="20px"
                      />
                      Entre com Google
                    </div>
                  </button>
                </div>
                <div>
                  <div
                    class="inline"
                  >
                    <div
                      class="divider"
                    />
                    <span
                      class="text"
                    >
                      Ou entre com o e-mail
                    </span>
                    <div
                      class="divider"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="xjgm6d4"
              style="--xjgm6d0: 16px;"
            >
              <div>
                <div
                  class="sc-egkSDF jFpkDV"
                >
                  <span
                    class="splu5g7 splu5g5"
                    data-component-name="Text2"
                    style="--splu5g0: 0.875rem; --splu5g2: 1.250rem; --splu5g1: 1.000rem; --splu5g3: 1.500rem; hyphens: auto; font-weight: 500; text-transform: inherit; text-decoration: inherit; overflow-wrap: anywhere; white-space: pre-line;"
                  >
                    E-mail
                  </span>
                  <input
                    class="sc-fAUdSK lbNUdd"
                    name="myInput"
                    placeholder="Insira seu e-mail"
                    type="text"
                  />
                </div>
              </div>
              <div>
                <div
                  class="sc-egkSDF jFpkDV"
                >
                  <span
                    class="splu5g7 splu5g5"
                    data-component-name="Text2"
                    style="--splu5g0: 0.875rem; --splu5g2: 1.250rem; --splu5g1: 1.000rem; --splu5g3: 1.500rem; hyphens: auto; font-weight: 500; text-transform: inherit; text-decoration: inherit; overflow-wrap: anywhere; white-space: pre-line;"
                  >
                    Senha
                  </span>
                  <div
                    class="xjgm6d4"
                    style="--xjgm6d0: 4px;"
                  >
                    <div>
                      <input
                        class="sc-fAUdSK lbNUdd"
                        name="myInput"
                        placeholder="Digite a sua senha"
                        type="text"
                      />
                    </div>
                    <div>
                      <div
                        class="kkfbnv5 kkfbnv8 kkfbnv9"
                        data-component-name="Inline"
                        style="--kkfbnv0: space-between; align-items: center;"
                      >
                        <div>
                          <div
                            class="checkboxContainer"
                          >
                            <div
                              aria-checked="false"
                              aria-labelledby=":r0:"
                              class="_1rz9b12 _1rz9b11 _1y2v1nfhn"
                              data-component-name="Checkbox"
                              role="checkbox"
                              tabindex="0"
                            >
                              <div
                                class="kkfbnv5 kkfbnv8 kkfbnv4"
                                data-component-name="Inline"
                                style="--kkfbnv0: 16px; align-items: stretch;"
                              >
                                <div>
                                  <div
                                    class="splu5g7 splu5g5"
                                    data-component-name="Text3"
                                    style="--splu5g0: 1.000rem; --splu5g2: 1.500rem; --splu5g1: 1.125rem; --splu5g3: 1.500rem; hyphens: auto; font-weight: 400; text-transform: inherit; text-decoration: inherit; overflow-wrap: anywhere; white-space: pre-line;"
                                  >
                                    <div
                                      style="position: relative; top: -2px;"
                                    >
                                      <div
                                        class="_1rz9b17 _1rz9b15 _1y2v1nfhl _1y2v1nfht _1y2v1nfhz _1y2v1nfi2 _1y2v1nfke _1y2v1nf3p"
                                      >
                                        <svg
                                          class="_1rz9b1a _1y2v1nfhm"
                                          height="10"
                                          viewBox="0 0 14 10"
                                          width="14"
                                        >
                                          <path
                                            d="M5 10L0 5.192l1.4-1.346L5 7.308 12.6 0 14 1.346z"
                                            fill="var(--_1vqcj1i1u)"
                                          />
                                        </svg>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                            <h3>
                              Lembre-se de mim
                            </h3>
                          </div>
                        </div>
                        <div>
                          <a
                            class="link"
                            href=""
                          >
                            Esqueceu a senha?
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <button
                  class="sc-ivxoEo duwQdc"
                >
                  Login
                </button>
              </div>
              <div>
                <div
                  class="kkfbnv5 kkfbnv8 kkfbnv4"
                  data-component-name="Inline"
                  style="--kkfbnv0: 8px; align-items: center;"
                >
                  <div>
                    <span
                      class="splu5g7 splu5g5"
                      data-component-name="Text1"
                      style="--splu5g0: 0.750rem; --splu5g2: 1.000rem; --splu5g1: 0.875rem; --splu5g3: 1.250rem; hyphens: auto; font-weight: 400; text-transform: inherit; text-decoration: inherit; overflow-wrap: anywhere; white-space: pre-line;"
                    >
                      Não tem conta?
                    </span>
                  </div>
                  <div>
                    <a
                      class="link"
                      href=""
                    >
                      Registre-se
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        class="allRights"
      >
        @2024 Böe Todos os direitos reservados
      </span>
    </div>
    <div
      class="sc-gtLWhw hxSQND"
    >
      <div
        class="_1xgjmkvc"
        style="--_1xgjmkv2: 48px; --_1xgjmkv1: 48px; --_1xgjmkv0: 48px; --_1xgjmkv5: 0px; --_1xgjmkv4: 0px; --_1xgjmkv3: 0px; --_1xgjmkv8: 48px; --_1xgjmkv7: 48px; --_1xgjmkv6: 48px; --_1xgjmkvb: 0px; --_1xgjmkva: 0px; --_1xgjmkv9: 0px; width: 70%; box-sizing: border-box;"
      >
        <div
          class="xjgm6d4"
          style="--xjgm6d0: 32px;"
        >
          <div>
            <img
              alt="boe-logo"
              src="[object Object]"
              width="70"
            />
          </div>
          <div>
            <div
              class="xjgm6d4"
              style="--xjgm6d0: 8px;"
            >
              <div>
                <h2>
                  O melhor para a saúde do seu gado
                </h2>
              </div>
              <div>
                <p>
                  Registre as análises da saúde dermatológica do seu gado e monitore todos os processos através do dashboard web.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  ;
</div>
`;
